src/libproc_macro/lib.rs

```diff
@@ -711,8 +711,8 @@ impl Term {
 
     /// Get a reference to the interned string.
     #[unstable(feature = "proc_macro", issue = "38356")]
-    pub fn as_str(&self) -> &str {
-        unsafe { &*(&*self.sym.as_str() as *const str) }
+    pub fn to_string(&self) -> String {
```

Comment:
If we go with this change I don't think we'd keep this as a method as it's already inherited through the `ToString` trait.